[AAP-5025] Audit view rule details page

lgalis opened the pull request:
Audit view rule details page
![Screenshot from 2022-12-06 03-46-59](https://user-images.githubusercontent.com/12769982/205866543-cda2bd55-f74a-43be-ac16-eea56a69392a.png)
![Screenshot from 2022-12-06 03-46-55](https://user-images.githubusercontent.com/12769982/205866548-9bf41b69-9f8b-4615-abda-ce1bef56f9c2.png)
![Screenshot from 2022-12-06 03-46-50](https://user-images.githubusercontent.com/12769982/205866550-6601340a-854a-48a0-ac92-f8181a985bc6.png)
![Screenshot from 2022-12-06 03-46-44](https://user-images.githubusercontent.com/12769982/205866553-d61b41cc-5f6a-4226-928a-12aa34887ea3.png)
![Screenshot from 2022-12-06 03-46-38](https://user-images.githubusercontent.com/12769982/205866554-6033c99e-eb27-44b3-b612-bfc0eba5f781.png)

lgalis committed:
Add audit rule details page

AlexSCorey commented on ui/src/app/routes.tsx:
```diff
@@ -234,11 +237,27 @@ const routes: AppRouteConfig[] = [
+  {
+    component: AuditHosts,
+    exact: true,
+    path: '/audit/hosts',
```
```suggestion
    path: '/audit-hosts',
```

lgalis replied:
same as above

AlexSCorey commented on ui/src/app/routes.tsx:
```diff
@@ -234,11 +237,27 @@ const routes: AppRouteConfig[] = [
     title: 'Audit View',
   },
+  {
+    component: AuditRules,
```
Remove this route

AlexSCorey commented on ui/src/app/routes.tsx:
```diff
@@ -234,11 +237,27 @@ const routes: AppRouteConfig[] = [
+  },
+  {
+    component: AuditRule,
+    path: '/audit-rule/:id',
```
```suggestion
    path: '/audit-rules/:id/details',
```

lgalis replied:
/audit-rule/:id is the defauult route for the details page

AlexSCorey commented on ui/src/test/AuditView/__snapshots__/audit-rules.test.tsx.snap:
```diff
@@ -0,0 +1,70 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`AuditHosts should render the AuditView component 1`] = `
```
When we started rewriting Controller we used snapshots in our tests.  However, not long after we removed them and abandoned them because they failed all the time, because we were developing and changing the views so quickly.  Because they failed so frequently we all ignored them, which prompted us to remove them. If this is the case for EDA I recommend removing them.

lgalis replied:
Snapshots are only as the first rounds of tests for the lists. Was planning to replace them with more detailed tests later. This no longer applies with the move to monorepo.

AlexSCorey commented on ui/src/test/AuditView/audit-hosts.test.tsx:
```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { mount } from 'enzyme';
```
We are moving away from enzyme because we have found that it is not keeping pace with the changes to react.  We are starting the process of adopting react testing, though if we move to a monorepo that might change.

lgalis replied:
Will not change the tests as we will be moving to th emonorepo

AlexSCorey commented on ui/src/test/AuditView/audit-rules.test.tsx:
```diff
@@ -0,0 +1,45 @@
+import { Route } from 'react-router-dom';
+import { mockApi } from '../__mocks__/baseApi';
+
+const ComponentWrapper = ({ children, initialEntries = ['/dashboard'] }) => (
```
Maybe we should move this out into a test utils file since it seems to be used frequently.

lgalis replied:
Will not change the tests as we will be moving to the monorepo

AlexSCorey reviewed (changes requested):
This is really good work and very exciting.  Please do not get overwhelmed with my comments.  Much of this is suggestions and conversations I'd like to have.  All the changes suggested do not have to be made in this PR, however, I do think that we need to remove the `console.logs` and update the urls.

lgalis commented:
@AlexSCorey - this PR is still a draft, was not yet submitted for review. Will make the requested changes.TY!

lgalis commented on ui/src/app/API/Audit.ts:
```diff
@@ -0,0 +1,29 @@
+import { getAxiosInstance } from '@app/API/baseApi';
+import { AxiosResponse } from 'axios';
+
+const auditRulesEndpoint = '/api/rules';
```
@these were for mocks only - the PR is a draft

lgalis committed:
Update audit rule details, rule hosts, rule jobs and rule events tabs

lgalis commented on ui/src/app/AuditView/audit-rule-jobs.tsx:
```diff
@@ -0,0 +1,148 @@
+    stateDispatch({ type: 'setFetching', payload: true });
+    return listAuditRuleJobs(rule?.id, pagination)
+      .then(() => stateDispatch({ type: 'setFetching', payload: false }))
+      .catch(() => stateDispatch({ type: 'setFetching', payload: false }));
```
I'll leave this as is here - and have a separate PR to make this change in all places it applies to.

lgalis commented on ui/src/app/AuditView/AuditView.tsx:
```diff
@@ -1,13 +1,69 @@
+        <Route exact path="/audit/hosts">
+          <AuditHosts/>
+        </Route>
+        <Route path="/audit">
```
I agree we need to have similar routes. I do like the /audit/hosts better - but let;s discuss at the next huild meeting - withe the move to the platform ui - we'll use the same routes as agreed for that project.

lgalis commented:
@AlexSCorey  - thank you fr the review, it is appreciated. It was a draft PR at the time of your review, though. I always remove all the console logs before marking the PR as ready for review. 
I addressed most of the comments above.

lgalis committed:
Merge branch 'main' into AAP-5025_audit_view_rule_detail

lgalis committed:
Merge branch 'main' into AAP-5025_audit_view_rule_detail

ddonahue007 reviewed (approved):
LGTM

lgalis committed:
Merge branch 'main' into AAP-5025_audit_view_rule_detail

lgalis committed:
Merge branch 'main' into AAP-5025_audit_view_rule_detail

Dostonbek1 reviewed (approved):
lgtm
